dask_cudf/io/csv.py

```diff
@@ -16,12 +17,20 @@ def read_csv(path, chunksize="256 MiB", **kwargs):
         path, tokenize, **kwargs
     )  # TODO: get last modified time
 
+    compression = kwargs.pop('compression', False)
     meta = cudf.read_csv(filenames[0], **kwargs)
 
     dsk = {}
     i = 0
     for fn in filenames:
         size = os.path.getsize(fn)
+        if chunksize and compression:
+            warn("Warning %s compression does not support breaking apart files\n"
+                "Please ensure that each individual file can fit in memory and\n"
+                "use the keyword ``blocksize=None to remove this message``\n"
+                "Setting ``chunksize=(size of file)``" % compression)
+        chunksize = size
```

Comment:
This line is probably intended to be indented.

Also, we can't trust `os.path.getsize` in this case because the file is compressed.  

Instead, I recommend a separate codepath for the case when `chunksize=None` that looks like the code before the byte_range PR.  This will be a simpler graph with one task per file that doesn't manage the keywords at all.

Reply:
moved the code path out the the for loop in last commit